rof-scheduler/src/SharedComponents/AddUser/index.js

```diff
@@ -1,281 +1,412 @@
-    const [loading, setLoading] = useState(false);
-    const [validationMap, setValidationMap] = useState(new Map());
-    const [errMsg, setErrMsg] = useState(undefined);
-    const [successMsg, setSuccessMsg] = useState(false);
-    const [disableBtns, setDisableBtns] = useState(false);
-    
-    const closeModal = function(){
-        setValidationMap(new Map());
-        setErrMsg(undefined);
-        handleHide();
-    }
+function AddUserModal({ userType, show, handleHide, handleUserAddSuccess }) {
+  const [loading, setLoading] = useState(false);
+  const [validationMap, setValidationMap] = useState(new Map());
+  const [errMsg, setErrMsg] = useState(undefined);
+  const [successMsg, setSuccessMsg] = useState(false);
+  const [disableBtns, setDisableBtns] = useState(false);
 
-    const handleCreateUser = function(e){
-        e.preventDefault();
+  const closeModal = function () {
+    setValidationMap(new Map());
+    setErrMsg(undefined);
+    handleHide();
+  };
 
-        setErrMsg(undefined);
-        var firstName = e.target.firstName.value;
-        var lastName = e.target.lastName.value;
-        var ssn = e.target.ssn.value;
-        var role = e.target.role.value;
-        var username = e.target.username.value;
-        var email = e.target.email.value;
-        var phoneNumber = e.target.phoneNumber.value;
-        var addressLine1 = e.target.addressLine1.value;
-        var addressLine2 = e.target.addressLine2.value;
-        var city = e.target.city.value;
-        var state = e.target.state.value;
-        var zipCode = e.target.zipCode.value;
-        var password = e.target.password.value;
-        var retypedPassword = e.target.retypedPassword.value;
-        var inputValidations = new Map();
+  const handleCreateUser = function (e) {
+    e.preventDefault();
 
-        //validate employee
-        if (userType === "Employee"){
-            inputValidations = ensureCreateEmployeeInformationProvided(
-                firstName,
-                lastName,
-                ssn,
-                role,
-                username,
-                email,
-                phoneNumber,
-                addressLine1,
-                city,
-                state,
-                zipCode,
-                password,
-                retypedPassword
-            );
-        }else{
-            //validate client
-        }
+    setErrMsg(undefined);
+    var firstName = e.target.firstName.value;
+    var lastName = e.target.lastName.value;
+    var username = e.target.username.value;
+    var email = e.target.email.value;
+    var addressLine1 = e.target.addressLine1.value;
+    var addressLine2 = e.target.addressLine2.value;
+    var city = e.target.city.value;
+    var state = e.target.state.value;
+    var zipCode = e.target.zipCode.value;
+    var password = e.target.password.value;
+    var retypedPassword = e.target.retypedPassword.value;
+    if (userType === "Employee") {
+      var ssn = e.target.ssn.value;
```

Comment:
this is not good. when you do this, technically ssn, role, and phonenumber only exist in the scope of this if statement. you cannot access it outside the if statement. try it out with another language like c# and they will throw you a compiler error.

instead, declare var ssn = undefined...etc outside if statement. then populate it inside if statement. that way, ssn, role, and phonenumber scope is within method and everything inside this method can access it.